src/app/core/draft/draft.service.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { AppConfig } from '../../app.config';
+import { CaseEventData } from '../../shared/domain/case-event-data';
+import { Observable } from 'rxjs/Observable';
+import { Response, ResponseOptions } from '@angular/http';
+import createSpyObj = jasmine.createSpyObj;
+import { DraftService } from './draft.service';
+import { Draft } from '../../shared/domain/draft';
+import { CaseDetails } from '../../shared/domain/case-details';
+
+describe('Drafts Service', () => {
+
+  const API_URL = 'http://aggregated.ccd.reform';
+  const JID = 'TEST';
+  const CT_ID = 'TestAddressBookCase';
+  const DRAFT_ID = 'Draft#1';
+  const EVENT_TRIGGER_ID = 'createCase';
+  const DRAFT_URL = API_URL + `/caseworkers/:uid/jurisdictions/${JID}/case-types/${CT_ID}/event-trigger/${EVENT_TRIGGER_ID}/drafts/`;
+  const ERROR: HttpError = new HttpError();
+  ERROR.message = 'Critical error!';
+
+  let appConfig: any;
+  let httpService: any;
+  let errorService: any;
+
+  let draftService: DraftService;
+
+  beforeEach(() => {
+    appConfig = createSpyObj<AppConfig>('appConfig', ['getApiUrl', 'getDraftUrl']);
+    appConfig.getApiUrl.and.returnValue(API_URL);
+    appConfig.getDraftUrl.and.returnValue(API_URL);
+
+    httpService = createSpyObj<HttpService>('httpService', ['get', 'post', 'put']);
+    errorService = createSpyObj<HttpErrorService>('errorService', ['setError']);
+
+    draftService = new DraftService(httpService, appConfig, errorService);
+  });
+
+  describe('saveDraft()', () => {
+    const CASE_EVENT_DATA: CaseEventData = {
+      event: {
+        id: EVENT_TRIGGER_ID,
+        summary: 'Short summary',
+        description: 'A very nice description'
+      },
+      event_token: 'test-token',
+      ignore_warning: false
+    };
+
+    const CASE_DETAILS: CaseDetails = {
+      id: 'string',
+      jurisdiction: 'Test Jurisdiction',
+      case_type_id: 'TestCaseType',
+      state: 'State 1',
+      case_data: {}
+    };
+
+    const DRAFT_RESPONSE: Draft = {
+      id: '5',
+      document: CASE_DETAILS,
+      type: 'Case Details',
+      created: '21-06-2018',
+      updated: '23-04-2019'
+    };
+
+    beforeEach(() => {
+      httpService.post.and.returnValue(Observable.of(new Response(new ResponseOptions({
+        body: JSON.stringify(DRAFT_RESPONSE)
+      }))));
+      httpService.put.and.returnValue(Observable.of(new Response(new ResponseOptions({
+        body: JSON.stringify(DRAFT_RESPONSE)
+      }))));
+      httpService.get.and.returnValue(Observable.of(new Response(new ResponseOptions({
+        body: JSON.stringify(DRAFT_RESPONSE)
+      }))));
+    });
+
+    // it('should get drafts on server', () => {
+    //   draftService
+    //     .getDrafts(JID, CT_ID)
+    //     .subscribe(
+    //       data => expect(data).toEqual(DRAFT_RESPONSE)
+    //     );
+    //   expect(httpService.get).toHaveBeenCalledWith(DRAFT_URL);
+    // });
+    //
+    // it('should set error when error is thrown when creating draft', () => {
```

Comment:
This one when uncommented needs to change name.

Reply:
Should we remove it and keep it on a branch for the get endpoint or stashed locally?

Reply:
If it should stay it should not be under saveDraft() wrapper.